python/pylibcugraph/pylibcugraph/graphs.pyx

```diff
@@ -414,3 +354,152 @@ cdef class MGGraph(_GPUGraph):
     def __dealloc__(self):
         if self.c_graph_ptr is not NULL:
             cugraph_mg_graph_free(self.c_graph_ptr)
+
+
+cdef class SGGraph_From_CSR(_GPUGraph):
```

Comment:
I don't think we should create a new graph type in order to support a different graph input format, since the resulting graph is ultimately the same. I think we should just provide a factory function that takes CSR arrays and returns a standard SGGraph, or extend the SGGraph ctor to support either COO or CSR arrays.

One proposal could be to re-implement the function named `construct_graph_from_csr()` as a cython `cdef` function that takes the CSR arrays and simply calls the C API function `cugraph_sg_graph_create_from_csr` to create a SGGraph.